src/Countries/Spain.php

```diff
@@ -6,35 +6,157 @@
 
 class Spain extends Country
 {
+    protected function __construct(
+        protected ?string $region = null,
+    ) {
+    }
+
     public function countryCode(): string
     {
         return 'es';
     }
 
-    /** @return array<string, CarbonImmutable> */
     protected function allHolidays(int $year): array
     {
-        return array_merge([
-            'Año Nuevo' => '01-01',
-            'Epifanía del Señor' => '01-06',
-            'Día del Trabajador' => '05-01',
-            'Asunción de la Virgen' => '08-15',
-            'Fiesta Nacional de España' => '10-12',
-            'Todos los Santos' => '11-01',
-            'Día de la Constitución Española' => '12-06',
-            'Inmaculada Concepción' => '12-08',
-            'Navidad' => '12-25'
-        ], $this->variableHolidays($year));
+        return array_merge(
+            [
+                'Año Nuevo' => '01-01',
+                'Epifanía del Señor' => '01-06',
+                'Día del Trabajador' => '05-01',
+                'Asunción de la Virgen' => '08-15',
+                'Fiesta Nacional de España' => '10-12',
+                'Todos los Santos' => '11-01',
+                'Día de la Constitución Española' => '12-06',
+                'Inmaculada Concepción' => '12-08', // 2024?
+                'Navidad' => '12-25'
+            ],
+            $this->variableHolidays($year),
+            $this->regionHolidays($year),
+        );
     }
 
     /** @return array<string, CarbonImmutable> */
     protected function variableHolidays(int $year): array
     {
-        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))
-            ->setTimezone('Europe/Madrid');
+        $easter = $this->easter($year);
 
         return [
             'Viernes Santo' => $easter->subDays(2),
         ];
     }
+
+    /** @return array<string, string> */
+    protected function regionHolidays(int $year): array
+    {
+        if ($year === 2024) {
+            return $this->regionHolidays2024();
+        }
+
+        return [];
```

Comment:
I'm not familiar with the holidays in Spain, but isn't it possible to calculate the days instead of hardcoding them? This way we can have correct data for more years.